app/client/components/filterModule/queryConstructor.tsx

```diff
@@ -90,9 +111,12 @@ class QueryConstructor extends React.Component<Props, ComponentState> {
   };
 
   onSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    this.setState({ searchString: e.target.value }, () => {
+    if (this.state.timeout) {
+       clearTimeout(this.state.timeout);
+    }
+    this.setState({ searchString: e.target.value, timeout: setTimeout(() => {
       this.handleQueryChange();
-    });
+    }, 300)})
   };
```

Comment:
I added a small timeout so that it only searches once the user stops typing. Personally, I find this to be a better UX

Reply:
We can remove this if you don't agree @michielderoos, I'm not strongly opinionated